### PR TITLE
[Refactor](exec) refactor the pipline code

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

[Refactor](exec) refactor the pipline code remove some unless code in BE

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

File: be/src/pipeline/task_scheduler.h [hunk 1]
```diff
@@ -66,8 +66,8 @@ class TaskScheduler {
 private:
     std::unique_ptr<ThreadPool> _fix_thread_pool;
     std::shared_ptr<TaskQueue> _task_queue;
-    std::vector<std::unique_ptr<std::atomic<bool>>> _markers;
-    std::atomic<bool> _shutdown;
+    std::vector<bool> _markers;
+    bool _shutdown;
```
Comment (yiguolei) [inline be/src/pipeline/task_scheduler.h:70]: need init this variable

Comment (HappenLee) [reply be/src/pipeline/task_scheduler.h:70]: image
construct func do the work

**Comment by HappenLee** [comment]
run buildall

File: be/src/pipeline/task_scheduler.cpp [hunk 2]
```diff
@@ -101,8 +98,7 @@ void _close_task(PipelineTask* task, PipelineTaskState state, Status exec_status
 }
 
 void TaskScheduler::_do_work(size_t index) {
-    const auto& marker = _markers[index];
-    while (*marker) {
+    while (_markers[index]) {
```
Comment (yiguolei) [inline be/src/pipeline/task_scheduler.cpp:101]: 这里语义似乎跟之前不一样。
原来是相当于 *_markers[index]

File: be/src/pipeline/task_scheduler.h (hunk 1 above)
Comment (yiguolei) [inline be/src/pipeline/task_scheduler.h:70]: 需要初始化

File: be/src/pipeline/task_scheduler.cpp (hunk 2 above)
Comment (HappenLee) [reply be/src/pipeline/task_scheduler.cpp:101]: 一样的，不影响结果正确

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Review by Gabriel39 — APPROVED** [review]

**Review by yiguolei — APPROVED** [review]

**Pull request merged**